recipes/_omnibus_toolchain.rb

```diff
@@ -17,64 +17,60 @@
+      basedir=default
+    EOH
+    owner 'root'
+    group 'root'
+    mode '0444'
+  end
+end
 
-  Chef::Log.info("`#{toolchain_name}` is already installed at version #{toolchain_version}.")
-else
-  install_options = ''
-  if solaris_10?
-    # create a nocheck file for automated install
-    file '/var/sadm/install/admin/auto-install' do
-      content <<-EOH.flush
-        mail=
-        instance=overwrite
-        partial=nocheck
-        runlevel=nocheck
-        idepend=nocheck
-        space=ask
-        setuid=nocheck
-        conflict=nocheckit
-        action=nocheck
-        basedir=default
-      EOH
-      owner 'root'
-      group 'root'
-      mode '0444'
-    end
+package_path = File.join(Chef::Config[:file_cache_path], File.basename(toolchain_url))
+version      = node['omnibus']['toolchain_version']
 
-    case node['kernel']['machine']
-    when 'i86pc'
-      package_url = "https://chef-releng.s3.amazonaws.com/omnibus/omnibus-toolchain/#{toolchain_name}-#{toolchain_version}-1.i86pc.solaris"
-    when 'sun4v', 'sun4u'
-      package_url = "https://chef-releng.s3.amazonaws.com/omnibus/omnibus-toolchain/#{toolchain_name}-#{toolchain_version}-1.sun4v.solaris"
-    end
+remote_file package_path do
+  source toolchain_url
+  action :create_if_missing
+end
 
-    install_options = '-a auto-install'
-  elsif aix?
-    package_url = "http://chef-releng.s3.amazonaws.com/omnibus/omnibus-toolchain/#{toolchain_name}-#{toolchain_version}-1.powerpc.bff"
-  elsif nexus?
-    package_url = "http://chef-releng.s3.amazonaws.com/omnibus/omnibus-toolchain/#{toolchain_name}-#{toolchain_version}-1.nexus7.x86_64.rpm"
-  else
-    Chef::Application.fatal!("I don't know how to install #{node['omnibus']['toolchain_name']} on this platform!", 1)
-  end
+package node['omnibus']['toolchain_name'] do
+  provider Chef::Provider::Package::Yum if nexus?
+  source package_path
+  version version
+  options '-a auto-install' if solaris2?
+end
 
-  package_path = File.join(Chef::Config[:file_cache_path], File.basename(package_url))
+toolchain_shell = "/opt/#{node['omnibus']['toolchain_name']}/embedded/bin/bash"
 
-  remote_file package_path do
-    source package_url
-    action :create_if_missing
-  end
+# Ensure the toolchain's bash is added to acceptable shells list
+execute 'chsec_login_shell' do
+  command "chsec -f /etc/security/login.cfg -s usw -a 'shells=/bin/sh,/bin/bsh,/bin/csh,/bin/ksh,/bin/tsh,/bin/ksh93,/usr/bin/sh,/usr/bin/bsh,/usr/bin/csh,/usr/bin/ksh,/usr/bin/tsh,/usr/bin/ksh93,/usr/bin/rksh,/usr/bin/rksh93,/usr/sbin/uucp/uucico,/usr/sbin/sliplogin,/usr/sbin/snappd,/usr/bin/bash,#{toolchain_shell}'"
+  only_if { aix? }
+end
 
-  package node['omnibus']['toolchain_name'] do
-    provider Chef::Provider::Package::Yum if nexus?
-    source package_path
-    options install_options
-  end
+user node['omnibus']['build_user'] do
+  shell "/opt/#{node['omnibus']['toolchain_name']}/embedded/bin/bash"
+  action :create
+  not_if { solaris_10? }
 end
```

Comment:
Don't we set the shell as part of the `_user` recipe? Why this extra user resource?